### DISCUSSION TITLE
how can I get the full path for filename to display?

### DISCUSSION THREAD

**Discussion by grimm26** [nvim-lualine/lualine.nvim#658] [Configuration]
I'd like the full path for `filename`, not just the name, not an abbreviated path. How can I do this?

**Comment by fitrh** [comment]
```lua
{
  'filename',
  path = 2,
}
```

See `:h lualine-filename-component-options` for more

**Reply by grimm26** [reply]
Thank you!!

**Comment by ghost** [comment]
it doenst seem to work for me 

![image](https://user-images.githubusercontent.com/104435563/165529290-9f42a6d1-1e4d-442a-8572-c3f7ce1a4b4d.png)

**Reply by fitrh** [reply]
@AyayaFbiPls If you want to provide options for a component, it should be in table form `{ function_name, opts }`, in your case, it should be

```lua
lualine_c = { { 'filename', path = 2 } }
```

Providing a section with `{ 'filename', path=2 }` is interpreted as:

- Give me `filename` component
- Give me `path=2` component, which does not exist so it throws an error

**Reply by ghost** [reply]
I see, thank you.
